libs/oeth/history/src/components/HistoryCard.tsx

```diff
@@ -1,83 +1,92 @@
 import { useState } from 'react';
 
-import { Box, Button, Stack, Typography } from '@mui/material';
-import { Card } from '@origin/shared/components';
+import { Box, Button, Divider, Stack, Typography } from '@mui/material';
 import { useIntl } from 'react-intl';
 
-import { HistoryFilterButton } from './HistoryButton';
+import { graphqlClient } from '@origin/oeth/shared';
+import { useQuery } from '@tanstack/react-query';
+import { isAddressEqual } from 'viem';
+import { useAccount } from 'wagmi';
+import {
+  HistoryTableDocument,
+  HistoryTableQuery,
+  HistoryTableWithFiltersDocument,
+} from '../queries.generated';
+import { ExportData } from './ExportData';
+import { HistoryFilters } from './Filters';
 import { HistoryTable } from './HistoryTable';
 
-import type { ColumnFilter } from '@tanstack/react-table';
+const PAGE_SIZE = 20;
 
 export function HistoryCard() {
-  const [isConnected, setConnectionState] = useState(false);
+  const [page, setPage] = useState(0);
+  const [filters, setFilters] = useState<string[]>([]);
+  const { address, isConnected } = useAccount();
+
+  const { data, isFetching } = useQuery(
```

Comment:
let's use `.graphql` files and codegen for generating gql custom hooks

Reply:
I do use it, but I have two different queries and i need to switch between them depending if I have filters or not. so actually the queries are generated using codegen but i need to switch the query dynamically.